src/tests/decider-distro-tests-rs/tests/test.rs

```diff
@@ -119,3 +533,1006 @@ async fn test_decider_installed() {
+        });
+        // Reqs: blockNumber, getLogs and one of gasPrice, getTransactionCount and sendRawTransaction
+        // (try all to not depend on the order)
+        for step in 0..3 {
+            let (method, params) = recv_request.recv().await.unwrap();
+            let response = match method.as_str() {
+                "eth_blockNumber" => Ok(json!(to_hex(BLOCK_INIT))),
+                "eth_getLogs" => {
+                    let log = serde_json::from_value::<LogsReq>(params[0].clone()).unwrap();
+                    Ok(json!([TestApp::log_test_app2(
+                        DEAL_ID,
+                        BLOCK_NUMBER,
+                        log.topics[1].as_str()
+                    )]))
+                }
+                "eth_sendRawTransaction" => Err(error_value.clone()),
+                "eth_getTransactionCount" => Err(error_value.clone()),
+                "eth_gasPrice" => Err(error_value.clone()),
+                _ => panic!("unexpected method: {}", method),
+            };
+            send_response.send(response).unwrap();
+        }
+    }
+    wait_decider(&mut client).await;
+    let deals = get_joined_deals(&mut client).await;
+    assert!(
+        deals.is_empty(),
+        "since the registration failed, the deal must not be joined"
+    );
+
+    update_config(&mut client, &oneshot_config()).await.unwrap();
+    {
+        // Reqs: blockNumber, getLogs, gasPrice, getTransactionCount and sendRawTransaction
+        for _ in 0..5 {
+            let (method, params) = recv_request.recv().await.unwrap();
+            let response = match method.as_str() {
+                "eth_blockNumber" => json!(to_hex(BLOCK_NUMBER_LATER)),
+                "eth_getLogs" => {
+                    let log = serde_json::from_value::<LogsReq>(params[0].clone()).unwrap();
+                    assert!(log.from_block <= BLOCK_NUMBER);
+                    assert!(log.to_block <= BLOCK_NUMBER);
+                    json!([TestApp::log_test_app1(
+                        DEAL_ID,
+                        BLOCK_NUMBER,
+                        log.topics[1].as_str()
+                    ),])
+                }
+                "eth_sendRawTransaction" => {
+                    // TODO: how not to wait for the registration if Decider failed?
+                    json!("0x55bfec4a4400ca0b09e075e2b517041cd78b10021c51726cb73bcba52213fa05")
+                }
+                "eth_getTransactionCount" => json!("0x1"),
+                "eth_gasPrice" => json!("0x3b9aca07"),
+                _ => panic!("unexpected method: {}", method),
+            };
+            send_response.send(Ok(response)).unwrap();
+        }
+    }
+    wait_decider(&mut client).await;
+    let deals = get_joined_deals(&mut client).await;
+    assert!(!deals.is_empty(), "the deal must be joined after fail")
+}
+
+/*
+// Update doesn't work and we don't know how it will work in future
+// Maybe, make this test ALWAYS fail to remind that this doesn't work?
```

Comment:
I think we can implement this test. Return a `DealChanged` log from RPC mock, and check that all the logic works.

Having this feature working _almost_ correctly is enough. Fundamental things won't change, and for the details we can tweak as they appear.

Same holds for Deal removal.

Reply:
can you create a task for that plz?

Reply:
well, I fixed CID encoding, but it still doesn't work with real chains. I created a task not to forget that it's broken https://linear.app/fluence/issue/NET-573/fix-deal-update-in-chain-connector

I kinda don't like that the tests on update are working and IRL it doesn't work